components/MapComponent.tsx

```diff
@@ -23,83 +32,142 @@ interface Props<T> {
     longitudeDelta: number
   }
 
-  markers: mapMarker[]
+  initialRadius: {
+    radius: number
+  }
+
+  markers: MapMarker[]
+  customizers: MarkerCustomize[]
+
+  currentSelectedMarker: String | undefined
 
   movementSettings: {
     canScroll: boolean
     canZoom: boolean
     canRotate: boolean
   }
+}
 
-  extractKey: (event: any) => string
+// When clicking on the marker, zoom towards where it is.
+// Using stackOverflow formula
+// stackoverflow.com/questions/5206018/fit-mapview-to-circle-bounds
+function circleCoordinateGeneration (lat: number, long: number, radius: number) {
```

Comment:
This could also use a better name (and some tests to ensure we don't mess up this somewhat obscure calculation if we ever need to configure it later)